### PR TITLE
Feature/vue3 upgrade

### PR DESCRIPTION
**Pull request by b3n4kh**
Upgrade vue to vue3.

Solves: #10 #37 #38 #54

### CODE REVIEW

File: src/gui/src/components/assess/CardNewsItem.vue [hunk 1]
```diff
@@ -117,11 +89,11 @@
...
+                  v-ripple="false"
                   class="mr-1 mt-1"
                   outlined
                   :to="'/newsitem/' + newsItem.id"
-                  v-on:click.stop
-                  v-ripple="false"
+                  @click.stop
                 >
                   <v-icon>mdi-eye</v-icon>
                   <span>view Details</span>
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/gui/src/components/assess/CardNewsItem.vue]: Will be fixed with #98

File: src/gui/src/views/users/settings/UserSettings.vue [hunk 2]
```diff
@@ -0,0 +1,175 @@
+<template>
```
Comment (ewoiset) [inline src/gui/src/views/users/settings/UserSettings.vue:1]: Functionality not given 
`UserSettings.vue:69 [intlify] Not found 'settings.open_item_source' key in 'en' locale messages.`

File: src/gui/src/views/nav/AssessNav.vue [hunk 3]
```diff
@@ -1,11 +1,11 @@
 <template>
   <filter-navigation
-    :search="filter.search"
+    :search="search"
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/gui/src/views/nav/AssessNav.vue:3]: Fixed in: https://github.com/ait-cs-IaaS/Taranis-NG/commit/8c5fbd11945225a740da8a6b2f8627e0d292b9fc

File: src/gui/src/components/analyze/ReportItem.vue [hunk 4]
```diff
@@ -40,48 +41,46 @@
...
-                :error-messages="errors.collect('title')"
               ></v-text-field>
             </v-col>
           </v-row>
           <v-row>
-            <v-col cols="12" class="pa-0 ma-0" v-if="report_type">
+            <v-col v-if="report_type && edit" cols="12" class="pa-0 ma-0">
               <v-expansion-panels
-                class="mb-1"
                 v-for="attribute_group in report_type.attribute_groups"
```
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/components/analyze/ReportItem.vue:62]: @b3n4kh  `is type === 'RICH_TEXT' ` in AttributeItem needed? if not, I will remove entire <vue-editor> scope

[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/components/analyze/ReportItem.vue:62]: see https://github.com/ait-cs-IaaS/Taranis-NG/issues/97

File: src/gui/src/views/users/settings/UserSettings.vue (hunk 2 above)
Comment (ewoiset) [reply src/gui/src/views/users/settings/UserSettings.vue:1]: see https://github.com/ait-cs-IaaS/Taranis-NG/issues/99

File: src/gui/src/views/admin/OSINTSourcesView.vue [hunk 5]
```diff
@@ -1,52 +1,54 @@
...
         <ImportExport @import="importData" @export="exportData"></ImportExport>
       </template>
     </DataTable>
     <EditConfig
       v-if="formData && Object.keys(formData).length > 0"
-      :configData.sync="formData"
-      :formFormat.sync="formFormat"
+      v-model:config-data="formData"
+      v-model:form-format="formFormat"
       @submit="handleSubmit"
```
[the remark this answers is not in the document]
Comment (b3n4kh) [reply src/gui/src/views/admin/OSINTSourcesView.vue:23]: Should be fixed now

File: src/gui/src/views/nav/AnalyzeNav.vue [hunk 6]
```diff
@@ -1,16 +1,16 @@
 <template>
   <filter-navigation
-    :search="filter.search"
-    @update:search="(value) => (search = value)"
+    :search="reportFilter.search"
```
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/views/nav/AnalyzeNav.vue:3]: if you select "today", then select "all", not everything is dispalyed. --> all has to be deselected to display all items.

File: src/gui/src/views/users/settings/UserSettings.vue (hunk 2 above)
Comment (ewoiset) [reply src/gui/src/views/users/settings/UserSettings.vue:1]: Language selection works perfectly, however after selecting a language the "save" button throws an error --> 

`marshmallow.exceptions.ValidationError: {'language': ['Unknown field.']}`

@b3n4kh  Maybe we can update this in the #99 issue

Comment (ewoiset) [reply src/gui/src/views/users/settings/UserSettings.vue:1]: Should Hotkeys be selectable?  current behaviour: `Not found 'settings.open_item_source' key in 'de' locale messages.`

is '@/views/users/settings/UserWordLists.vue' not found realated?

File: src/gui/src/views/admin/OSINTSourcesView.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/views/admin/OSINTSourcesView.vue:23]: Items can be added and deleted, however table is not refreshing

File: src/gui/src/views/users/AssessView.vue
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/views/users/AssessView.vue:1]: This is a wanted  (new) behavior

File: src/gui/src/views/nav/AnalyzeNav.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (ewoiset) [reply src/gui/src/views/nav/AnalyzeNav.vue:3]: Offset not set correctly in UI

**Pull request merged**